src/test/java/io/camunda/testing/assertions/IncidentAssertTest.java

```diff
@@ -0,0 +1,449 @@
+
+@ZeebeAssertions
+class IncidentAssertTest {
+
+  public static final String WRONG_VALUE = "wrong value";
+  public static final String ERROR_CODE = "error";
+  public static final String ERROR_MESSAGE = "error occurred";
+
+  private ZeebeClient client;
+  private ZeebeEngine engine;
+
+  // These tests are for testing assertions as well as examples for users
+  @Nested
+  class HappyPathTests {
+
+    private RecordStreamSource recordStreamSource;
+
+    @Test
+    void testHasErrorType() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      incidentAssert.hasErrorType(ErrorType.UNHANDLED_ERROR_EVENT);
+    }
+
+    @Test
+    void testHasErrorMessage() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      incidentAssert.hasErrorMessage(
+          "Expected to throw an error event with the code 'error' with message 'error occurred', but it was not caught. No error events are available in the scope.");
+    }
+
+    @Test
+    void testExtractErrorMessage() throws InterruptedException {
+      // given
+      deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode(ERROR_CODE)
+          .errorMessage(ERROR_MESSAGE)
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      final StringAssert messageAssert = incidentAssert.extractErrorMessage();
+
+      // then
+      Assertions.assertThat(messageAssert).isNotNull();
```

Comment:
🙃 `extractErrorMessage()` will never return null

Reply:
Well, you know that because you looked in the implementation. But the method could return null.